src/crimson/os/seastore/cache.cc

```diff
@@ -20,8 +20,11 @@ using std::string_view;
 
 namespace crimson::os::seastore {
 
-Cache::Cache(SegmentManager &segment_manager) :
-  segment_manager(segment_manager)
+Cache::Cache(
+  ExtentReader &reader,
+  SegmentManager& segment_manager)
```

Comment:
I don't think we need a segment_manager instance here.  The only remaining user is prepare_record, which calls get_block_size().  I think block size needs to be global throughout seastore and will be set at mkfs time.  Can you add it to the superblock and pass it in explicitly to Cache instead of the SegmentManager instance?

Reply:
For now, you can just pass in the block_size to Cache.  We can worry about putting the block size into the superblock later.